webui/src/pages/setup/communicationPreferences.tsx

```diff
@@ -0,0 +1,76 @@
+import {Error} from "../../lib/components/controls";
+
+interface CommunicationPreferencesSetupProps {
+    onSubmit: (email: string, updatesCheck: boolean, securityCheck: boolean) => Promise<void>;
+    setupError: Error;
+    disabled: boolean;
+}
+
+export const CommunicationPreferencesSetup: FC<CommunicationPreferencesSetupProps> = ({
+    onSubmit,
+    setupError,
+    disabled,
+}) => {
+    const [userEmail, setUserEmail] = useState<string>("");
+    const [updatesCheck, setUpdatesCheck] = useState<boolean>(false);
+    const [securityCheck, setSecurityCheck] = useState<boolean>(false);
+
+    const submitHandler = useCallback((e) => {
+        onSubmit(userEmail, updatesCheck, securityCheck);
+        e.preventDefault();
+    }, [onSubmit, userEmail, updatesCheck, securityCheck]);
+
+    const handleEmailChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+        setUserEmail(e.target.value);
+    }, [setUserEmail]);
+
+    const handleUpdatesChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+        setUpdatesCheck(e.target.checked);
+    }, [setUpdatesCheck]);
+
+    const handleSecurityChange = useCallback((e: ChangeEvent<HTMLInputElement>) => {
+        setSecurityCheck(e.target.checked);
+    }, [setSecurityCheck]);
+
+    return (
+        <Row>
+            <Col md={{offset: 2, span: 8}}>
+                <Card className="setup-widget">
+                    <Card.Header>Communication Preferences</Card.Header>
+                    <Card.Body>
+                        <Card.Text>
+                            Providing your email address is optional.
+                            We recommend at least signing up for security updates. <br />
+                            You may unsubscribe from either or both at any time.
+                        </Card.Text>
+                        <Form onSubmit={submitHandler}>
+                            <Form.Group controlId="user-email" className="mt-4">
+                                <Form.Label>Email (optional)</Form.Label>
+                                <Form.Control type="email" placeholder="name@company.com" value={userEmail}  onChange={handleEmailChange} autoFocus/>
+                            </Form.Group>
+
+                            <Form.Group controlId="updates-check" className="mt-4">
+                                <Form.Label>News & feature updates</Form.Label>
+                                <Form.Check type="checkbox" checked={updatesCheck} onChange={handleUpdatesChange} label="I'd like to receive news and feature updates" />
+                            </Form.Group>
+
+                            <Form.Group controlId="security-check" className="mt-4">
+                                <Form.Label>Security updates</Form.Label>
+                                <Form.Check type="checkbox" checked={securityCheck} onChange={handleSecurityChange} label="I'd like to receive security updates" />
+                            </Form.Group>
```

Comment:
I think it's better to have the security updates subscription checked by default

Reply:
At first, I had it checked by default, but it's more aligned with current regulations to leave it unchecked. Checking the boxes is a more clear-cut opt-in action.  
If they want to receive these updates, they'll check the box.